sycl/CMakeLists.txt

```
@@ -71,130 +71,23 @@ set(SYCL_INCLUDE_DIR "include")
-    list(APPEND AUX_CMAKE_FLAGS -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER})
-  endif()
-  file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/icd_build)
-  set(OpenCL_ICD_LOADER_SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/OpenCL/icd")
-  if (NOT DEFINED SYCL_EP_OCL_ICD_SKIP_AUTO_UPDATE)
-    set(SYCL_EP_OCL_ICD_SKIP_AUTO_UPDATE ${SYCL_EXTERNAL_PROJECTS_SKIP_AUTO_UPDATE})
-  endif()
-  ExternalProject_Add(ocl-icd
-    GIT_REPOSITORY    https://github.com/KhronosGroup/OpenCL-ICD-Loader.git
-    GIT_TAG           5f8249691ec8c25775789498951f8e9eb62c201d
-    UPDATE_DISCONNECTED ${SYCL_EP_OCL_ICD_SKIP_AUTO_UPDATE}
-    SOURCE_DIR        ${OpenCL_ICD_LOADER_SOURCE_DIR}
-    BINARY_DIR        "${CMAKE_CURRENT_BINARY_DIR}/icd_build"
-    INSTALL_DIR       "${CMAKE_CURRENT_BINARY_DIR}/icd_install"
-    CMAKE_ARGS -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-               -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
-               -DOPENCL_ICD_LOADER_HEADERS_DIR=${OpenCL_INCLUDE_DIRS}
-               -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
-               -DCMAKE_INSTALL_LIBDIR:PATH=lib${LLVM_LIBDIR_SUFFIX}
-               ${AUX_CMAKE_FLAGS}
-    STEP_TARGETS      configure,build,install
-    DEPENDS           ocl-headers
-    BUILD_BYPRODUCTS ${OpenCL_LIBRARIES}
-    LOG_DOWNLOAD 1
-    LOG_UPDATE 1
-    LOG_CONFIGURE 1
-    LOG_BUILD 1
-    LOG_INSTALL 1
-  )
-  ExternalProject_Add_Step(ocl-icd llvminstall
-    COMMAND ${CMAKE_COMMAND} -E copy_directory <INSTALL_DIR>/ ${LLVM_BINARY_DIR}
-    COMMENT "Installing ocl-icd into the LLVM binary directory"
-    DEPENDEES install
-  )
-  # Optionally install the ICD library along with LLVM.
-  option(OpenCL_INSTALL_KHRONOS_ICD_LOADER
-    "Install the Khronos ICD Loader built as part of LLVM" OFF)
-  if (OpenCL_INSTALL_KHRONOS_ICD_LOADER)
-    install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/icd_install/"
-      DESTINATION "."
-      COMPONENT opencl-icd
-    )
-  endif()
-else()
-  file(GLOB ICD_LOADER_SRC "${OpenCL_LIBRARIES}*")
-  file(COPY ${ICD_LOADER_SRC} DESTINATION ${LLVM_LIBRARY_OUTPUT_INTDIR})
-  add_custom_target(ocl-icd DEPENDS ${OpenCL_LIBRARIES} COMMENT "Copying OpenCL ICD Loader ...")
-endif()
-
-set(OPENCL_INCLUDE "${OpenCL_INCLUDE_DIRS}")
-
-add_library (OpenCL-Headers INTERFACE)
-add_library (OpenCL::Headers ALIAS OpenCL-Headers)
-target_include_directories(OpenCL-Headers
-  INTERFACE ${OPENCL_INCLUDE}
-)
-install(DIRECTORY ${OPENCL_INCLUDE}/CL
-  DESTINATION ${SYCL_INCLUDE_DEPLOY_DIR}/sycl
-  COMPONENT opencl-headers
-)
+# LLVM build sets this OFF by default, but we need OpenCL to be built as shared
+# library.
+set(BUILD_SHARED_LIBS ON)
+# second argument is not used, it's required since first arg is not a real
+# subdirectory
+add_subdirectory(../opencl ${CMAKE_CURRENT_BINARY_DIR}/opencl_deps)
+
+# Copy OpenCL Headers into sycl headers build directory
+# Compiler does automatic lookup bin/../include based on clang binary location,
+# e.g. when run LIT tests
+file(COPY ${OPENCL_ICD_LOADER_HEADERS_DIR}/CL
```

```suggestion
file(COPY ${OpenCL_INCLUDE_DIRS}/CL
```
They are not really OPENCL ICD loader headers but OpenCL headers. We can use standard variable instead.

---

OPENCL_ICD_LOADER_HEADERS_DIR is a variable from ICD loader CMakeLists.txt (see opencl/CMakeLists.txt line 26) and it actually points to OpenCL headers as a dependency. I'm setting it up and using globally as cache var for uniformity. 
We should decide what to do with OpenCL_INCLUDE_DIRS first anyway.

---

I think OpenCL_INCLUDE_DIR should be reused to avoid confusion.

---

Done. Also cleaned up.